tracer/test/Datadog.Trace.TestHelpers/MockTracerAgent.cs

```diff
@@ -823,10 +827,33 @@ private void HandleHttpRequests()
                             }
 
                             var response = HandleHttpRequest(MockHttpParser.MockHttpRequest.Create(ctx.Request));
-                            var buffer = Encoding.UTF8.GetBytes(response);
+                            var buffer = behaviour == AgentBehaviour.WRONG_ANSWER ? Encoding.UTF8.GetBytes("WRONG DATA") : Encoding.UTF8.GetBytes(response);
+
+                            if (!ctx.Request.RawUrl.ToLower().Contains("/traces"))
+                            {
+                                if (behaviour == AgentBehaviour.RETURN_404)
+                                {
+                                    ctx.Response.StatusCode = 404;
+                                }
+
+                                if (behaviour == AgentBehaviour.RETURN_500)
+                                {
+                                    ctx.Response.StatusCode = 500;
+                                }
+                            }
```

Comment:
I like the overall approach here, but I think this behaviour should be inside `HandleHttpRequest`. That way we
 
- can easier detect the "traces" endpoint to special-case it
- can test it with other transports (this is TCP only, we need to test named pipes and UDS too)

Reply:
I have moved the test logic to the HandleHttpRequest method so we could test also the other transports